awx/main/tasks/receptor.py

```diff
@@ -411,11 +413,16 @@ def _run_internal(self, receptor_ctl):
                     return
 
                 try:
-                    resultsock = receptor_ctl.get_work_results(self.unit_id, return_sockfile=True)
-                    lines = resultsock.readlines()
-                    receptor_output = b"".join(lines).decode()
-                    if receptor_output:
-                        self.task.runner_callback.delay_update(result_traceback=receptor_output)
+                    if state_name == 'Failed' and self.task.runner_callback.event_ct == 0:
+                        # if receptor work unit failed and no events were emitted, work results may
+                        # contain useful information about why the job failed. In case stdout is
+                        # massive, only ask for last 1000 bytes
+                        startpos = max(stdout_size - 1000, 0)
+                        resultsock = receptor_ctl.get_work_results(self.unit_id, startpos=startpos, return_sockfile=True)
+                        lines = resultsock.readlines()
+                        receptor_output = b"".join(lines).decode()
+                        if receptor_output:
+                            self.task.runner_callback.delay_update(result_traceback=receptor_output)
```

Comment:
I'm concerned that in this case we may lose the info from `detail`, where `receptor_output` is empty

Reply:
I fixed it up so that we still get Detail in the case where there are no results